mirrord/cli/src/operator.rs

```diff
@@ -149,158 +145,17 @@ async fn get_status_api(config: Option<&Path>) -> CliResult<Api<MirrordOperatorC
-                "Original Target",
-                "Namespace",
-                "Copy Pod Name",
-                "Scale Down?"
-            ]);
-
-            for (pod_name, copy_target_resource) in copy_targets {
-                copy_targets_table.add_row(row![
-                    copy_target_resource.spec.target.to_string(),
-                    copy_target_resource
-                        .metadata
-                        .namespace
-                        .as_deref()
-                        .unwrap_or_default(),
-                    pod_name,
-                    if copy_target_resource.spec.scale_down {
-                        "*"
-                    } else {
-                        ""
-                    },
-                ]);
-            }
-
-            copy_targets_table.printstd();
-        }
-        println!();
-    }
-
-    if let Some(statistics) = status.statistics.as_ref() {
-        println!("Operator Daily Users: {}", statistics.dau);
-        println!("Operator Monthly Users: {}", statistics.mau);
-    }
-
-    let mut sessions = Table::new();
-
-    sessions.add_row(row![
-        "Session ID",
-        "Target",
-        "Namespace",
-        "User",
-        "Ports",
-        "Session Duration"
-    ]);
-
-    for session in &status.sessions {
-        let locked_ports = session
-            .locked_ports
-            .as_deref()
-            .map(|ports| {
-                ports
-                    .iter()
-                    .map(|(port, type_, filter)| {
-                        format!(
-                            "Port: {port}, Type: {type_}{}",
-                            filter
-                                .as_ref()
-                                .map(|f| format!(", Filter: {}", f))
-                                .unwrap_or_default()
-                        )
-                    })
-                    .collect::<Vec<_>>()
-                    .join("\n")
-            })
-            .unwrap_or_default();
-
-        sessions.add_row(row![
-            session.id.as_deref().unwrap_or(""),
-            &session.target,
-            session.namespace.as_deref().unwrap_or("N/A"),
-            &session.user,
-            locked_ports,
-            humantime::format_duration(Duration::from_secs(session.duration_secs)),
-        ]);
-    }
-
-    sessions.printstd();
-
-    Ok(())
-}
+pub(super) mod status;
 
 /// Handle commands related to the operator `mirrord operator ...`
 pub(crate) async fn operator_command(args: OperatorArgs) -> CliResult<()> {
     match args.command {
         OperatorCommand::Setup(params) => operator_setup(params).await.map_err(CliError::from),
-        OperatorCommand::Status { config_file } => operator_status(config_file.as_deref()).await,
+        OperatorCommand::Status { config_file } => {
+            StatusCommandHandler::new(config_file)
+                .and_then(StatusCommandHandler::handle)
+                .await
+        }
```

Comment:
It's just to use the same pattern as we have for the sessions handler.

Now for the pattern itself, I find it neater than just using a free function, because we separate initialization of stuff from the code that is doing the actual work. We could also break it down further into steps, so if you needed to add something, say we had "stepfyied" this, then adding the sqs status would follow the pattern of: add new method, and call it like `Command::new().copy_targets().active_sessions().sqs().print()`, where these steps are interchangeable, except for `new` and `print`.